### PR TITLE
#16715 Adding new folder reserved words: dotAdmin and custom_elements

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

File: dotCMS/src/main/resources/dotmarketing-config.properties [hunk 1]
```diff
@@ -272,7 +272,7 @@ path.to.dojo=/html/js/dojo/custom-build
 URLMAP_FALLTHROUGH=true
 
 ##	Reserved Folder Names
-RESERVEDFOLDERNAMES=WEB-INF,META-INF,assets,dotcms,html,portal,email_backups,DOTLESS,DOTSASS
+RESERVEDFOLDERNAMES=WEB-INF,META-INF,assets,dotcms,html,portal,email_backups,DOTLESS,DOTSASS,dotAdmin,custom_elements
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/resources/dotmarketing-config.properties]: I don't think it's safe to put these restrictions in the .properties file, as customers might unintentionally change it without really knowing the consequences of it. They should only be available as they are in the `FolderAPIImpl` class.

Comment (wezell) [reply dotCMS/src/main/resources/dotmarketing-config.properties]: I am ok if we remove this from property file and have the default values only as defaults in the class using the overloaded `Config.getStringProperty` methods.  This hides the config so people don't mess with it unless they know what they are doing and want to override the property, at which point that is on them.

Comment (dsilvam) [reply dotCMS/src/main/resources/dotmarketing-config.properties]: so @nollymar we need to remove the property from the `dotmarketing-config.properties`

File: dotCMS/src/integration-test/resources/it-dotmarketing-config.properties [hunk 2]
```diff
@@ -317,7 +317,7 @@ path.to.dojo=/html/js/dojo/custom-build
 URLMAP_FALLTHROUGH=true
 
 ##	Reserved Folder Names
-RESERVEDFOLDERNAMES=WEB-INF,META-INF,assets,dotcms,html,portal,email_backups,DOTLESS,DOTSASS
+RESERVEDFOLDERNAMES=WEB-INF,META-INF,assets,dotcms,html,portal,email_backups,DOTLESS,DOTSASS,dotAdmin,custom_elements
```
Comment (dsilvam) [inline dotCMS/src/integration-test/resources/it-dotmarketing-config.properties]: remove this property from the properties file

**Review by dsilvam — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java [hunk 3]
```diff
@@ -92,7 +92,7 @@
 				CollectionsUtils.set(Config.getStringArrayProperty("RESERVEDFOLDERNAMES",
 					new String[]{"WEB-INF", "META-INF", "assets", "dotcms", "html", "portal",
 							"email_backups",
-							"DOTLESS", "DOTSASS"})
+							"DOTLESS", "DOTSASS", "dotAdmin", "custom_elements"})
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java]: test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java (hunk 3 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java]: tests for this logic have already been implemented :)

File: dotCMS/src/integration-test/resources/it-dotmarketing-config.properties (hunk 2 above)
Comment (nollymar) [reply dotCMS/src/integration-test/resources/it-dotmarketing-config.properties]: done

File: dotCMS/src/main/resources/dotmarketing-config.properties (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/main/resources/dotmarketing-config.properties]: done

**Review by freddyucv — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**